packages/docs/Examples/sale.md

```diff
@@ -3,20 +3,48 @@ order: -40
 icon: flame
 ---
 
-# Sale
+# Sales
 
-In the previous example both smart objects as well as their owners had to be known before the swap transaction could be built. In this section we show how a sale offer can be built by a seller with needing to know the buyer in advance.
+In the case of a swap, both smart objects have to be known before the swap transaction can be built. In this section we show how a single smart object can be offered for sale at a specified price.
 
-## Smart Contract
+## Sale
 
-We will be reusing the `Swap` smart contract from above, but we also need a new smart contract `Payment` below. Buyer will use this contract to create a smart object holding the satoshis to pay for the NFT
+We first explain how to execute the sale of a smart object. The smart contract is quite simple but the swap does not preserving ordinal ranges of the smart object being sold. It is therefore not safe to use with ordinals. In the next section we explain a slightly more complicated swap that can be used with ordinals. 
+
+### Smart Contract
 
 ```ts
-class Payment extends Contract {
-  _owners: string[]
+class Sale extends Contract {
+  static exec(a: NFT, b: NFT) {
+    const [ownerA] = a._owners
+    const [ownerB] = b._owners
+    a.transfer(ownerB)
+    b.transfer(ownerA)
+    return [b, a]
+  }
+}
+```
+
+#### Sighash Types
+
+The idea is to create a transaction with two inputs and two outputs. The inputs spend the smart object `a` to be sold as well as another smart object `b` that is used for the payment. Like in the case of a swap the smart contract exchanges the owners of the smart objects. However unlike the swap smart contract it returns `[b, a]`. This has the effect that the first output of the transaction will represent `b` after the call and the second will represent `a`.
+
+Seller then builds the swap transaction and signs it with the [sighash type](https://developer.bitcoin.org/devguide/transactions.html#signature-hash-types) `SIGHASH_SINGLE | SIGHASH_ANYONECANPAY` and the input index `0`. This means that Sellers signature remains valid even when arbitrary inputs and outputs are added to the transaction. In particular, another user could change or remove the Bitcoin Computer specific meta data that is located in output 3.
+
+Essentially Seller is saying: you can do with this inputs-output pair whatever you want as long as you spend the output indicated in the first input and you keep the locking script and the amount of the first output the same. Seller can confidentially publish such a partially signed transaction as he wants to sell the nft spent by the fist inputs for the amount indicated in the first output and both are signed.
```

Comment:
Can we rephrase this somehow? An idea could be: Essentially Seller is saying: you are at liberty to utilize the provided input-output pair in any manner, under the condition that you expend the output specified in the initial input and maintain consistency with both the locking script and the amount allocated to the first output. Seller can confidentially publish such a partially signed transaction, since the sighash type used to sign the input-output pair warranties to sell the nft for the amount indicated on the first output.